chain/block.go

```diff
@@ -290,6 +294,46 @@ func (b *StatelessBlock) initializeBuilt(
 			b.containsWarp = true
 		}
 	}
+
+	// transaction hash generation
+	db, err := merkledb.New(ctx, memdb.New(), merkledb.Config{
```

Comment:
It would be great to create a helper for this inside of `HyperSDK` so this root generation technique could be used more easily in other places.

Reply:
But this is the right direction 👍